test/plugins/TPSetBufferCreator.cpp

```diff
@@ -0,0 +1,336 @@
+      if (first) {
+        TLOG() << get_name() << ": Got first TPSet, with start_time=" << input_tpset.start_time
+               << " and end_time=" << input_tpset.end_time;
+        first = false;
+      }
+
+      if (m_tps_buffer->add(input_tpset)) {
+        ++addedCount;
+        // TLOG() << "TPSet start_time=" << input_tpset.start_time << " and end_time=" << input_tpset.end_time<< " added
+        // ("<< addedCount << ")! And buffer size is: " << m_tps_buffer->get_stored_size() << " /
+        // "<<m_tps_buffer->get_buffer_size();
+      }
+      else{
+        ++addFailedCount;
+      }
+
+      if (m_dr_on_hold.size()) { // check if new data is part of data request on hold
+
+        // TLOG() << "On hold DRs: "<<m_dr_on_hold.size();
+
+        std::map<dfmessages::DataRequest, std::vector<trigger::TPSet>>::iterator it = m_dr_on_hold.begin();
+
+        while (it != m_dr_on_hold.end()) {
+          // TLOG() << "Checking TPSet (sart_time= "<<input_tpset.start_time <<", end_time= "<< input_tpset.end_time <<"
+          // w.r.t. DR on hold ("<< it->first.window_begin  <<", "<< it->first.window_end  <<"). TPSet count:
+          // "<<it->second.size();
+          if ((it->first.window_begin < input_tpset.end_time && it->first.window_begin > input_tpset.start_time) ||
+              (it->first.window_end > input_tpset.start_time && it->first.window_end < input_tpset.end_time) ||
+              (it->first.window_end > input_tpset.end_time &&
+               it->first.window_begin < input_tpset.start_time)) { // new tpset is whithin data request windown?
+            it->second.push_back(input_tpset);
+            // TLOG() << "Adding TPSet (sart_time="<<input_tpset.start_time <<" on DR on hold ("<<
+            // it->first.window_begin  <<", "<< it->first.window_end  <<"). TPSet count: "<<it->second.size();
+          }
+          if (it->first.window_end < input_tpset.start_time) { // If more TPSet aren't expected to arrive then push and remove peding data request
+            requested_tpset.txsets_in_window = it->second;
+            std::unique_ptr<dataformats::Fragment> frag_out = convert_to_fragment(requested_tpset, it->first);
+            TLOG() << get_name() << ": Sending late requested data (" << (it->first).window_begin << ", "
+                   << (it->first).window_end << "), containing " << requested_tpset.txsets_in_window.size()
+                   << " TPSets.";
+            if (!requested_tpset.txsets_in_window.size()) {
+              frag_out->set_error_bit(dataformats::FragmentErrorBits::kDataNotFound, true);
+            }
+
+            send_out_fragment(std::move(frag_out), sentCount, running_flag);
+            m_dr_on_hold.erase(it);
+            it--;
+            continue;
+          }
+          it++;
+        }
+      } // end if(m_dr_on_hold.size())
+
+    } catch (const dunedaq::appfwk::QueueTimeoutExpired& excpt) {
+    }
+
+    // Block that reveives data requests and return fragments from buffer
```

Comment:
Small typo: reveives to reviews?

Reply:
Fixed, see 5a6d5b9820511087ecdd00a053bc520a9e0e0ea5